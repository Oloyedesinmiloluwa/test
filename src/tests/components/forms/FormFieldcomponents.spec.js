import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import FormComponents from '../../../components/forms/FormFieldComponents';

describe(' LoginModal component rendering', () => {
  const spy = sinon.spy();
  const wrapper = shallow(<FormComponents
    type="email"
    name="email"
    onChange={spy}
    value="email"
    placeholder="enter password"
    label="username"
    errors="Invalid Input"
  />);

  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should handle and store the email input', () => {
    wrapper.find('#email').simulate('change', {
      target: { name: 'email', value: 'nwokeochavictor@gmail.com' },
    });
    expect(spy.called).toBe(true);
  });
});
