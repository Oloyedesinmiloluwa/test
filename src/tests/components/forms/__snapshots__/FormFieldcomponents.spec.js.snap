// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` LoginModal component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormField
    errors="Invalid Input"
    label="username"
    name="email"
    onChange={[Function]}
    placeholder="enter password"
    type="email"
    value="email"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          htmlFor="email"
        >
          username
          <input
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="enter password"
            type="email"
            value="email"
          />
        </label>,
        <Label
          color="red"
          pointing={true}
        >
          Invalid Input
        </Label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "username",
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="enter password"
              type="email"
              value="email"
            />,
          ],
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": Array [
          "username",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "enter password",
              "type": "email",
              "value": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Invalid Input",
          "color": "red",
          "pointing": true,
        },
        "ref": null,
        "rendered": "Invalid Input",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            htmlFor="email"
          >
            username
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="enter password"
              type="email"
              value="email"
            />
          </label>,
          <Label
            color="red"
            pointing={true}
          >
            Invalid Input
          </Label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "username",
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="enter password"
                type="email"
                value="email"
              />,
            ],
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": Array [
            "username",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "enter password",
                "type": "email",
                "value": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Invalid Input",
            "color": "red",
            "pointing": true,
          },
          "ref": null,
          "rendered": "Invalid Input",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
